src/main/java/utilities/SaucelabsDriverConfiguration.java

```diff
@@ -18,8 +19,73 @@
 import org.openqa.selenium.edge.EdgeOptions;
 import org.openqa.selenium.firefox.FirefoxOptions;
 import org.openqa.selenium.remote.RemoteWebDriver;
+import saucelabs.api.ApiUtils;
+import saucelabs.api.Response;
+import saucelabs.client.SauceLabsClient;
+import saucelabs.dto.AppStorageItemMetadataResponse;
+import saucelabs.dto.AppStorageResponse;
+import saucelabs.service.SauceLabsService;
+
+public class SaucelabsDriverConfiguration {
+
+  private static final String USER = LocalEnviroment.getUser();
+  private static final String ACCESS_TOKEN = LocalEnviroment.getAccessToken();
+  private static final String AUTHORIZATION =
+      Base64.getEncoder().encodeToString((USER + ":" + ACCESS_TOKEN).getBytes());
+
+  public static String getSaucelabsAppId(
+      String authorization, String appId, String kind, String version) {
+    // Check that appId is not null
+    if (appId.isBlank()) {
+      throw new IllegalArgumentException("AppIdentifier is not set");
+    }
+
+    // Retrieve app storage response
+    SauceLabsService sauceLabsService = new SauceLabsService(new SauceLabsClient());
+    Response<AppStorageResponse> response =
+        sauceLabsService.getV1StorageFiles(authorization, appId, kind.toLowerCase(), 10);
+    ApiUtils.checkStatusCode(response.getStatus(), SC_OK);
+    AppStorageResponse appStorageResponse = response.getPayload();
+
+    // Filter the apps
+    return appStorageResponse.getItems().stream()
+        .filter(item -> isValidApp(item.getMetadata(), version))
+        .findFirst()
+        .orElseThrow(() -> new IllegalArgumentException("Version not found: ".concat(version)))
+        .getId();
+  }
+
+  // This function abstracts the logic of getting the latest version of the app independent of the
+  // platform
+  private static String getVersion(AppStorageItemMetadataResponse metadata) {
+    String version;
+    if (LocalEnviroment.isAndroid()) {
+      version = metadata.getVersion();
+    } else {
+      version = metadata.getShort_version();
+    }
+    if (version == null) {
```

Comment:
return Objects.requireNonNull(version, "The app is not compatible with the set platform: ".concat(platform)");